File: c++/p4/main.cc
#include <iostream>
class Human {
    public:
    Human(){
        std::cout << "こにちわ、世界😇" << std::endl;
    }
    template <typename T> void girlfriend(){
        std::cout << "I don't have girlfriend" << std::endl;
    }
    template <typename T> void speak(T language){
        std::cout << "I can speak" << language << std::endl;
    }
    template <typename T> void run(T place){
        std::cout << "I run to" << place << std::endl;
    }
    private:
    template <typename T> void say(){
        std::cout << "愛してます💜" << std::endl;
    }
};
int main(){
    Human jame {};
    jame.speak<std::string>("English");
}

File: rust/p4/src/main.rs

trait paternal_grandfather{

}
trait paternal_grandmother{

}
trait maternal_grandfather{

}
trait maternal_grandmother{

}

trait father{
    fn run(){}
    fn read(){
        println!("I can read English");
    }
}
trait mother {
    fn speak(&self){
        println!("i can speak Japanese");
    }
    fn learn(){}
}
#[derive(Debug)]
struct son{
    name: &'static str,
    old: i32,
    height: f32,
}
impl mother for son{
    fn learn(){
        println!("I learn Rust");
    }
}
impl father for son{
    fn run(){
        println!("I run to Tokyo");
    }
}
fn main() {
    let s = son{name: "jame", old:20,height:165.0};
    s.speak();
}
